### PR TITLE
Add Enterprise license clarifications to 23.2 release notes

### PR DESCRIPTION
**Pull request by mikeCRL**
I'm making some updates to clarify Enterprise license requirements, as we've done in past major release notes, per a suggestion from Shannon.

For review in the [preview page](https://deploy-preview-18243--cockroachdb-docs.netlify.app/docs/releases/v23.2.html) or [diff](https://github.com/cockroachdb/docs/pull/18243/files):

1. New note about Enterprise license being required only for features where that's specified.
2. Addition of Enterprise license requirement to Column Level Encryption. (It was already specified in the linked docs.)
3. Questions in [comment](https://github.com/cockroachdb/docs/pull/18243/files#r1465590808) on the existing PCR entry around expressing the type of license.
4. Whether any other listed features that do not mention Enterprise actually do require it. (Note: I stumbled upon this issue about [Read Committed](https://cockroachlabs.atlassian.net/browse/CRDB-35379) and wasn't sure if it meant the feature does currently require the license.)

**Updated:**
- [releases/v23.2.md](https://deploy-preview-18243--cockroachdb-docs.netlify.app/docs/releases/v23.2.html)
- [src/current/v23.2/enterprise-licensing.md](https://deploy-preview-18243--cockroachdb-docs.netlify.app/docs/v23.2/enterprise-licensing.html)
  - (These tables are also included at 
[src/current/v23.2/get-started-with-enterprise-trial.md](https://deploy-preview-18243--cockroachdb-docs.netlify.app/docs/v23.2/get-started-with-enterprise-trial.html))

### CODE REVIEW

File: src/current/_includes/releases/v23.2/v23.2.0.md [hunk 1]
```diff
@@ -188,8 +192,8 @@ This section summarizes the most significant user-facing changes in v23.2.0 and
     <!-- Rows for each feature -->
     <tr>
     <td>
-        <p class="feature-summary">Physical Cluster Replication is now available in a public preview</p>
-        <p class="feature-description"><a href="https://www.cockroachlabs.com/docs/v23.2/physical-cluster-replication-overview">Physical Cluster Replication</a> is an asynchonous replication feature that allows your cluster to recover from full-cluster failure with a low RPO and RTO. In 23.2, it is an Enterprise-only Public Preview feature, requiring a CCL license, and only available for self-hosted CockroachDB deployments.</p>
+        <p class="feature-summary">Physical Cluster Replication is now available in Preview</p>
+        <p class="feature-description"><a href="https://www.cockroachlabs.com/docs/v23.2/physical-cluster-replication-overview">Physical Cluster Replication</a> is an asynchonous replication feature that allows your cluster to recover from full-cluster failure with a low RPO and RTO. In 23.2, it is an Enterprise-only Preview feature, requiring a CCL license, and only available for self-hosted CockroachDB deployments.</p>
```
Comment (mikeCRL) [inline src/current/_includes/releases/v23.2/v23.2.0.md:196]:
```suggestion
        <p class="feature-summary">Physical Cluster Replication is now available in Preview</p>
        <p class="feature-description"><a href="https://www.cockroachlabs.com/docs/v23.2/physical-cluster-replication-overview">Physical Cluster Replication</a> is an asynchonous replication feature that allows your cluster to recover from full-cluster failure with a low RPO and RTO. In 23.2, it is a Preview feature, requiring an Enterprise license, and only available for self-hosted CockroachDB deployments.</p>
```

I'd like to make this edit but I'm not sure about dropping the CCL mention. As-is, "CCL license" seems ambiguous - on our FAQ we refer to "CCL (Free)" vs " CCL (Paid)". Regardless, is it sufficient to mention an "Enterprise license" as we do elsewhere in the docs?

**Comment by mikeCRL** [comment]
@kannanlakshmi @awoods187 @mwang1026 I'd appreciate your review. Please see the list of proposed changes and an open question at the [top](https://github.com/cockroachdb/docs/pull/18243#issue-2099155461) of this PR.

**Comment by kannanlakshmi** [comment]
Thanks for including me, Mike. I'll defer to Andy and Michael on license changes for DB.

**Review by awoods187 — APPROVED** [review]

File: src/current/_includes/releases/v23.2/v23.2.0.md (hunk 1 above)
Comment (shannonbradshaw) [reply src/current/_includes/releases/v23.2/v23.2.0.md:196]: @mikeCRL let's make this change as you suggest. The other wording is confusing.

**Review by shannonbradshaw — APPROVED** [review]
Thanks, Mike!

**Comment by shannonbradshaw** [comment]
Actually, we need to update this page as well: https://www.cockroachlabs.com/docs/stable/enterprise-licensing

**Comment by mikeCRL** [comment]
Additional changes made to catch up the feature list on https://www.cockroachlabs.com/docs/stable/enterprise-licensing.

Merging. Thanks, all, for the help.

@mwang1026 when you get a chance, could you please see the second note on [this page](https://deploy-preview-18243--cockroachdb-docs.netlify.app/docs/releases/v23.2.html) (for awareness and any potential edits), and also let me know if we've missed flagging any other 23.2 features as Enterprise in the tables that follow?

**Pull request merged**